test/model/create.test.ts

```diff
@@ -46,6 +46,34 @@ test('creates a new entity with an array property', () => {
   expect(exactUser).toHaveProperty('arrayProp', [1, 2, 3])
 })
 
+test('creates a new entity with an array property with array of objects assigned', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      arrayProp: Array,
+    },
+  })
+
+  const arrayOfObjects = [
+    {
+      key: 'hobby',
+      value: 'code',
+    },
+    {
+      key: 'phone',
+      value: 12345,
+    },
+  ]
+
+  const exactUser = db.user.create({
+    id: 'abc-123',
+    arrayProp: arrayOfObjects,
+  })
+
+  expect(exactUser).toHaveProperty('id', 'abc-123')
+  expect(exactUser).toHaveProperty('arrayProp', arrayOfObjects)
```

Comment:
I recommend accessing the property directly on the target object instead of using `.toHaveProperty` method. One of the benefits of doing so is getting type validation when accessing a non-existing property (i.e. a typo, or when the property got removed). `.toHaveProperty` accepts a general `string` as the first argument. That string may not exist as a property on the target object. 

```suggestion
  expect(exactUser.arrayProp).toEqual(arrayOfObjects)
```

Reply:
Valid point as well! 

Initially, I wanted to make the test consistent with the convention that was already there, that's why I went with `toHaveProperty`. After adjusting the matchers for that case I've decided to apply it for other tests too (https://github.com/mswjs/data/pull/169/commits/b192a5bae0960eaf9557bb0a82732235555e22e8) - let me know if you have some concerns about that. 

I've noticed an issue with inferring the structure of the nested resource, that's why I went with https://github.com/mswjs/data/pull/169/commits/5cd820da591e8ceb794e6bf6438e00b6d994daa6#diff-9e87669b89f42eb50b386f8e76828b8c646e42c60598b3d5cd50d9b36152ba2bR179

Reply:
My apologies, the test suites are not yet aligned on this pattern. Yet another pull request idea to carry out 😉

> I've noticed an issue with inferring the structure of the nested resource
I think I've fixed that in #171 but that may be an entirely different issue.